Makefile

```diff
@@ -210,31 +214,10 @@ $(specs_dir)/examples/sum-to-n-spec.k: resources/sum-to-n.md $(TANGLER)
 # Casper
 casper_tmpls:=casper/module-tmpl.k casper/spec-tmpl.k
 
-$(specs_dir)/casper/recommended_target_hash-spec.k: $(casper_tmpls) casper/casper-spec.ini
+$(specs_dir)/casper/%-spec.k: $(casper_tmpls) casper/casper-spec.ini
 	@echo >&2 "==  gen-spec: $@"
 	mkdir -p $(dir $@)
-	python3 resources/gen-spec.py $^ recommended_target_hash recommended_target_hash > $@
-	cp casper/abstract-semantics.k $(dir $@)
-	cp casper/verification.k $(dir $@)
-
-$(specs_dir)/casper/proc_reward-spec.k: $(casper_tmpls) casper/casper-spec.ini
-	@echo >&2 "==  gen-spec: $@"
-	mkdir -p $(dir $@)
-	python3 resources/gen-spec.py $^ proc_reward proc_reward > $@
-	cp casper/abstract-semantics.k $(dir $@)
-	cp casper/verification.k $(dir $@)
-
-$(specs_dir)/casper/vote-spec.k: $(casper_tmpls) casper/casper-spec.ini
-	@echo >&2 "==  gen-spec: $@"
-	mkdir -p $(dir $@)
-	python3 resources/gen-spec.py $^ vote recommended_target_hash vote > $@
-	cp casper/abstract-semantics.k $(dir $@)
-	cp casper/verification.k $(dir $@)
-
-$(specs_dir)/casper/delete_validator-spec.k: $(casper_tmpls) casper/casper-spec.ini
-	@echo >&2 "==  gen-spec: $@"
-	mkdir -p $(dir $@)
-	python3 resources/gen-spec.py $^ delete_validator delete_validator > $@
+	python3 resources/gen-spec.py $^ $* $* > $@
```

Comment:
It will not work for `vote` spec, because `vote` requires `recommended_target_hash` and `proc_reward`. You can not use `$*`.